Add the fake initial snapshot to the snapshots.

nc6 opened the pull request:
This is needed to allow the initial stake pools to continue producing
blocks into the second epoch.

HERE BE DRAGONS: this function is not suitable for production use!

mrBliss commented:
> HERE BE DRAGONS: this function is not suitable for production use!

But now it *will be* used by any pure Shelley node. Or is that not "production"?

What are the consequences of using it?

The warning should really live in the code, not just in this commit message.

nc6 commented:
It will not be used unless `registerGenesisStaking` is invoked. But you're right, I should add this comment onto that function!

mrBliss commented:
> It will not be used unless `registerGenesisStaking` is invoked. But you're right, I should add this comment onto that function!

But `registerGenesisStaking` *is currently* called in `protocolInfoShelley`.

nc6 commented:
Hmm, true. OK, I have added a comment that explains the use, but I think a better solution would be to make this optional in `ShelleyGenesis` and then avoid doing any manipulation. I've added an issue at https://github.com/input-output-hk/cardano-ledger-specs/issues/1633

deepfire commented:
Ohh, but then pool block creation stops when the second epoch starts...

deepfire commented:
Works just fine now!

mrBliss commented on ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:
```diff
@@ -253,6 +253,10 @@ protocolInfoShelley genesis initialNonce maxMajorPV protVer mbCredentials =
+    --
+    -- HERE BE DRAGONS! This function is intended to help in testing. It should
+    -- not be called with anything other than 'emptyGenesisStaking' in
+    -- production.
```
So production /= Shelley testnets.

nc6 replied:
On shelley testnets it should be called with `emptyGenesisStaking`

deepfire commented:
bors r+

deepfire commented:
bors r+

nc6 commented:
bors retry

nc6 commented:
bors r+
